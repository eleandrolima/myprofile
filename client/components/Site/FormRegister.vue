<template>
  <form @submit.prevent="onSubmit">
    <CardBox id="card-box">
      <Input
        id="first-name"
        v-model="user.firstName"
        label="Nome"
        placeholder="Seu primeiro nome."
        :violations="constraint.getViolationsBy('firstName')"
      />

      <Input
        id="last-name"
        v-model="user.lastName"
        label="Sobrenome"
        placeholder="Seu segundo nome."
        :violations="constraint.getViolationsBy('lastName')"
      />

      <InputEmail
        id="email"
        v-model="user.email"
        :violations="constraint.getViolationsBy('email')"
      />

      <InputPassword
        v-model="user.password"
        :violations="constraint.getViolationsBy('password')"
      />

      <InputPassword
        id="repeat-password"
        v-model="user.confirmPassword"
        label="Confirme a senha"
        :violations="constraint.getViolationsBy('confirmPassword')"
      />

      <Button id="register-button" class="my-6" :loading="loading">
        Cadastrar
      </Button>

      <div id="ask-for-login" class="text-center text-sm mt-2">
        <span>JÃ¡ tem conta?</span>
        <NuxtLink to="/login" class="text-primary underline font-semibold">
          {{ $t('register') }}
        </NuxtLink>
      </div>
    </CardBox>
  </form>
</template>

<script>
import InputEmail from '../Form/InputEmail'
import InputPassword from '../Form/InputPassword'
import Button from '../Form/Button'
import Input from '../Form/Input'
import CardBox from '~/components/Site/CardBox'
import { ConstraintViolationListException } from '~/exception/constraint-violation-list.exception'
import { UserRegister } from '~/resources/user.register'
export default {
  components: { CardBox, Input, Button, InputPassword, InputEmail },
  data () {
    return {
      user: new UserRegister(),
      constraint: new ConstraintViolationListException(),
      loading: false
    }
  },
  methods: {
    async onSubmit () {
      try {
        this.loading = true
        await this.$userRepository.save(this.user)
        this.loading = false
      } catch (constraint) {
        this.constraint = constraint
        this.loading = false
      }
    }
  }
}
</script>
