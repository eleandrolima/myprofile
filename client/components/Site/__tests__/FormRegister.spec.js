import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import FormRegister from '../FormRegister'

function factory () {
  return shallowMount(FormRegister, {
    stubs: {
      NuxtLink: RouterLinkStub
    }
  })
}

describe('Form register component', () => {
  it('has all fields necessary to register new user.', () => {
    const wrapper = factory()

    expect(wrapper.find('#first-name').exists()).toBeTruthy()
    expect(wrapper.find('#last-name').exists()).toBeTruthy()
    expect(wrapper.find('#email').exists()).toBeTruthy()
    expect(wrapper.find('#password').exists()).toBeTruthy()
    expect(wrapper.find('#repeat-password').exists()).toBeTruthy()
    expect(wrapper.find('#register-button').exists()).toBeTruthy()
  })

  it('has a link to go to login page.', () => {
    const wrapper = factory()

    expect(wrapper.find('#ask-for-login').html()).toMatchSnapshot()
  })
})
