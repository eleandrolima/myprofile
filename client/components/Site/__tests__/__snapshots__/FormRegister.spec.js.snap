// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form register component has a link to go to login page. 1`] = `
<div id="ask-for-login" class="text-center text-sm mt-2"><span>JÃ¡ tem conta?</span> <a class="text-primary underline font-semibold">
    register
  </a></div>
`;
